pkg/scheduler/core/generic_scheduler.go

```diff
@@ -415,3 +416,52 @@ func (g *genericScheduler) divideReplicasAggregatedWithClusterReplicas(clusterAv
 	}
 	return targetClusters, nil
 }
+
+func (g *genericScheduler) ScaleSchedule(ctx context.Context, placement *policyv1alpha1.Placement,
+	object *workv1alpha1.ObjectReference, targetClusters []workv1alpha1.TargetCluster) (result ScheduleResult, err error) {
+	newTargetClusters := make([]workv1alpha1.TargetCluster, len(targetClusters))
+
+	if object.Replicas > 0 && placement.ReplicaScheduling != nil {
+		if placement.ReplicaScheduling.ReplicaSchedulingType == policyv1alpha1.ReplicaSchedulingTypeDuplicated {
+			for i, cluster := range targetClusters {
+				newTargetClusters[i] = workv1alpha1.TargetCluster{Name: cluster.Name, Replicas: object.Replicas}
+			}
+			result.SuggestedClusters = newTargetClusters
+			return result, nil
+		}
+		if placement.ReplicaScheduling.ReplicaSchedulingType == policyv1alpha1.ReplicaSchedulingTypeDivided {
+			preSelectedClusters := g.getPreSelected(targetClusters)
+			if placement.ReplicaScheduling.ReplicaDivisionPreference == policyv1alpha1.ReplicaDivisionPreferenceWeighted {
+				if placement.ReplicaScheduling.WeightPreference == nil {
+					return result, fmt.Errorf("no WeightPreference find to divide replicas")
+				}
+				clustersWithReplicase, err := g.divideReplicasByStaticWeight(preSelectedClusters, placement.ReplicaScheduling.WeightPreference.StaticWeightList, object.Replicas)
+				if err != nil {
+					return result, fmt.Errorf("failed to assignReplicas with Weight: %v", err)
+				}
+				result.SuggestedClusters = clustersWithReplicase
+				return result, nil
+			}
+		}
+	}
+
+	for i, cluster := range targetClusters {
+		newTargetClusters[i] = workv1alpha1.TargetCluster{Name: cluster.Name}
+	}
+	result.SuggestedClusters = newTargetClusters
+	return result, nil
+}
+
+func (g *genericScheduler) getPreSelected(targetClusters []workv1alpha1.TargetCluster) []*clusterv1alpha1.Cluster {
+	PreSelectedClusters := make([]*clusterv1alpha1.Cluster, len(targetClusters))
+	clusterInfoSnapshot := g.schedulerCache.Snapshot()
+	for i, targetCluster := range targetClusters {
+		for _, cluster := range clusterInfoSnapshot.GetClusters() {
+			if targetCluster.Name == cluster.Cluster().Name {
+				PreSelectedClusters[i] = cluster.Cluster()
+				break
+			}
+		}
+	}
+	return PreSelectedClusters
+}
```

Comment:
Yes, it will filter out these clusters

Reply:
`PreSelectedClusters := make([]*clusterv1alpha1.Cluster, len(targetClusters))`

this will always return a slice with length `len(targetClusters)`. If any cluster is disappeared, then you will get a `nil` slot,
that will trigger panic.

Reply:
My mistake。I will update this func。